src/mozilla_sec_eia/models/sec_eia_record_linkage/transform_sec_input.py

```diff
@@ -0,0 +1,383 @@
+    # if the subsidiary is matched to an SEC filer
+    ex21_df = create_sec_company_id_for_ex21_subs(ex21_df=ex21_df)
+    ex21_df = flatten_companies_across_time(
+        df=ex21_df, key_cols=["sec_company_id"], date_col="report_year"
+    )
+    ex21_df = ex21_df.fillna(np.nan)
+
+    return ex21_df
+
+
+def transform_basic10k_table(
+    basic_10k_df: pd.DataFrame, sec10k_filing_metadata: pd.DataFrame
+) -> pd.DataFrame:
+    """Transformations on SEC basic 10K filer table to prepare for record linkage."""
+    basic_10k_df = basic_10k_df.reset_index().pivot_table(
+        values="value", index="filename", columns="key", aggfunc="first"
+    )
+    basic_10k_df.columns.name = None
+    basic_10k_df = (
+        basic_10k_df.reset_index()
+        .pipe(_remove_weird_sec_cols)
+        .pipe(_add_report_year_to_sec, sec10k_filing_metadata)
+        .rename(columns=SEC_COL_MAP)
+        .pipe(clean_location_of_inc)
+        .pipe(transform_company_name)
+        .assign(
+            zip_code=lambda df: df["zip_code"].str[:5],
+            files_10k=True,
+            sec_company_id=lambda df: df["central_index_key"],
+        )
+        .pipe(fill_street_address_nulls)
+    )
+    basic_10k_df[STR_COLS] = basic_10k_df[STR_COLS].apply(
+        lambda x: x.str.strip().str.lower()
+    )
+    basic_10k_df["street_address"] = expand_street_name_abbreviations(
+        basic_10k_df["street_address"]
+    )
+    # flatten across time on unique company name and address pair
+    basic_10k_df = flatten_companies_across_time(
+        df=basic_10k_df, key_cols=["company_name", "street_address"]
+    )
+
+    return basic_10k_df
+
+
+@asset(
+    ins={
+        "basic_10k_dfs": AssetIn("basic_10k_company_info"),
+        # "clean_ex21_df": AssetIn("transformed_ex21_subsidiary_table"),
+        "sec10k_filing_metadata_dfs": AssetIn("sec10k_filing_metadata"),
+        # specify an io_manager_key?
+    },
+)
+def core_sec_10k__filers(
+    basic_10k_dfs: dict[str, pd.DataFrame],
+    # clean_ex21_df: pd.DataFrame,
+    sec10k_filing_metadata_dfs: dict[str, pd.DataFrame],
+) -> pd.DataFrame:
+    """Asset for creating an SEC 10K output table.
+
+    Flatten the table across time to only keep the most recent record
+    for each CIK. Add in Ex. 21 subsidiaries and link them to already present
+    filing companies. Create an sec_company_id for subsidiaries that aren't linked
+    to a CIK.
+    """
+    basic_10k_df = pd.concat(basic_10k_dfs.values())
+    sec10k_filing_metadata = pd.concat(sec10k_filing_metadata_dfs.values())
+    basic_10k_df = transform_basic10k_table(basic_10k_df, sec10k_filing_metadata)
+    # exclude Ex. 21 subs and just match to filers
+    # once the match has been conducted, add back in the Ex. 21 subs
+    out_df = basic_10k_df.fillna(np.nan).reset_index(names="record_id")
+    # TODO: Here we conduct the match to EIA and add on a column with utility_id_eia
```

Comment:
This is where I think it makes sense to call the splink model and run record linkage between the SEC 10K filers and the EIA utilities. Basically I think the transformed basic 10k table should be an input, and the output of the splink model is that basic 10K table with `utility_id_eia` as a column added on. In my head, I'm calling this matched asset `core_sec_10k__filers`, but maybe the whole way I'm thinking about this structure is wrong.